File: new/dart-maor/enrollment-paid-status_test.dart
import 'enrollment-paid-status.dart';

/// רתמת-זהב: אותן 6 דוגמאות-חוזה בדיוק מ-new/atoms/enrollment-paid-status.test.mjs.
/// שקעי-maor מקומיים (הבדיקה מייבאת רק את האטום שלה).
num paidOf(Map<String, dynamic> e) {
  final ps = e['payments'];
  final list = ps is List ? ps : const [];
  num a = 0;
  for (final p in list) {
    final amt = (p as Map)['amount'];
    a += (amt is num && amt.isFinite) ? amt : 0;
  }
  return a;
}

num payBal(Map<String, dynamic> e) {
  final td = e['totalDue'];
  final num due = td is num ? td : 0;
  final bal = due - paidOf(e);
  return bal > 0 ? bal : 0;
}

void main() {
  final c = <List<dynamic>>[
    [<String, dynamic>{'paidFull': true}, 'paid'],
    [<String, dynamic>{'totalDue': 500, 'payments': [{'amount': 300}, {'amount': 200}]}, 'paid'],
    [<String, dynamic>{'totalDue': 500, 'payments': [{'amount': 200}]}, 'partial'],
    [<String, dynamic>{'totalDue': 500, 'payments': <dynamic>[]}, 'unpaid'],
    [<String, dynamic>{'payments': [{'amount': 100}]}, 'unpaid'],
    [<String, dynamic>{'totalDue': 0, 'paidFull': false}, 'unpaid'],
  ];
  var f = 0;
  for (final row in c) {
    final e = row[0] as Map<String, dynamic>;
    final w = row[1] as String;
    final g = enrollmentPaidStatus(e, payBal, paidOf);
    if (g != w) {
      print('✗ $e ⇒ $g ≠ $w');
      f = 1;
    }
  }
  if (f != 0) throw StateError('enrollment-paid-status: סטייה מהמקור');
  print('✓ enrollment-paid-status: 6 דוגמאות-חוזה — ירוק');
}

File: new/atoms/enrollment-paid-status.test.mjs
import { enrollmentPaidStatus as __pure_enrollmentPaidStatus } from './enrollment-paid-status.mjs';
// צילום-מקומי + עטיפת-כריכה (מנוע-הטיהור v3)
const __d_enrollmentPaidStatus_ENROLLMENT_PAID_STATUS_T = {
  k1: "paid",
  k2: "partial",
  k3: "unpaid",
};
const enrollmentPaidStatus = (...a) => __pure_enrollmentPaidStatus(...a, ...Array(Math.max(0, 3 - a.length)).fill(undefined), __d_enrollmentPaidStatus_ENROLLMENT_PAID_STATUS_T);
// שקעים אמיתיים כסמנטיקת-maor (מקומיים לבדיקה — הבדיקה מייבאת רק את האטום שלה)
const paidOf = (e) => (e.payments || []).reduce((a, p) => a + (Number.isFinite(p.amount) ? p.amount : 0), 0);
const payBal = (e) => Math.max(0, (e.totalDue || 0) - paidOf(e));
const C = [
  [{ paidFull: true }, 'paid'],
  [{ totalDue: 500, payments: [{ amount: 300 }, { amount: 200 }] }, 'paid'],
  [{ totalDue: 500, payments: [{ amount: 200 }] }, 'partial'],
  [{ totalDue: 500, payments: [] }, 'unpaid'],
  [{ payments: [{ amount: 100 }] }, 'unpaid'],
  [{ totalDue: 0, paidFull: false }, 'unpaid'],
];
let f = 0;
for (const [e, w] of C) {
  const g = enrollmentPaidStatus(e, payBal, paidOf);
  if (g !== w) { console.error(`✗ ${JSON.stringify(e)} ⇒ ${g} ≠ ${w}`); f = 1; }
}
if (f) process.exit(1);
console.log('✓ enrollment-paid-status: 6 דוגמאות-חוזה — ירוק');
